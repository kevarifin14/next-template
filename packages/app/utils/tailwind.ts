export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
