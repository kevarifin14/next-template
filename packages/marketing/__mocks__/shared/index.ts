export * from 'shared';

export const mockPost = jest.fn();
export const post = mockPost;
