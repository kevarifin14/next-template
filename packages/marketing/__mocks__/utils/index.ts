import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
