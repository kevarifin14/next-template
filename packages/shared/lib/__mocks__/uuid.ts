export const mockUuid = '123456879';
export const v4 = jest.fn(() => mockUuid);
