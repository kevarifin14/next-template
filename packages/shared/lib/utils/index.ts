export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
