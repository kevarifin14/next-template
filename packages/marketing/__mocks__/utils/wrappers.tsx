import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
