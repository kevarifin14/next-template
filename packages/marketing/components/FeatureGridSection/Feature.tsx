import { CheckCircleIcon } from 'shared';

export type IFeature = {
  name: string,
  description: string,
};

type FeatureProps = {
  feature: IFeature
};

export default function Feature({ feature }: FeatureProps) {
  return (
    <div key={feature.name} className="relative">
      <dt className="flex space-x-2">
        <CheckCircleIcon />
        <h3 className="text-xl font-medium m-0">{feature.name}</h3>
      </dt>
      <dd className="mt-2 ml-8 text-base">{feature.description}</dd>
    </div>
  );
}
