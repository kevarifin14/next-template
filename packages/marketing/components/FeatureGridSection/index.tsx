import Section from 'components/Section';

import Feature, { IFeature } from './Feature';

type FeatureGridSectionProps = {
  title: string,
  description?: string,
  features: IFeature[],
};

export default function FeatureGridSection({
  title, description, features,
}: FeatureGridSectionProps) {
  return (
    <Section size="7xl">
      <div className="max-w-3xl mx-auto text-center">
        <h2 className="text-3xl font-extrabold text-gray-900">{title}</h2>
        <p className="mt-4 text-xl">
          {description}
        </p>
      </div>
      <dl className="mt-12 space-y-10 sm:space-y-0 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-12 lg:grid-cols-4 lg:gap-x-8">
        {features.map((feature) => <Feature key={feature.name} feature={feature} />)}
      </dl>
    </Section>
  );
}
