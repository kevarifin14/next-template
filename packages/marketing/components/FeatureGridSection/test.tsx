import { render } from '@testing-library/react';

import { mockDescription, mockFeatures, mockTitle } from '__mocks__/utils';

import FeatureGridSection from '.';

describe('FeatureGridSection', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(
      <FeatureGridSection
        title={mockTitle}
        description={mockDescription}
        features={mockFeatures}
      />,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
