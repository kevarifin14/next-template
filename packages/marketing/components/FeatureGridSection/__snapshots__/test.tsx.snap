// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureGridSection matches snapshot 1`] = `
<body>
  <div>
    <div
      class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl"
    >
      <div
        class="max-w-3xl mx-auto text-center"
      >
        <h2
          class="text-3xl font-extrabold text-gray-900"
        >
          mockTitle
        </h2>
        <p
          class="mt-4 text-xl"
        >
          mockDescription
        </p>
      </div>
      <dl
        class="mt-12 space-y-10 sm:space-y-0 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-12 lg:grid-cols-4 lg:gap-x-8"
      >
        <div
          class="relative"
        >
          <dt
            class="flex space-x-2"
          >
            <svg
              class="text-green-500 h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fill-rule="evenodd"
              />
            </svg>
            <h3
              class="text-xl font-medium m-0"
            >
              Marketing
            </h3>
          </dt>
          <dd
            class="mt-2 ml-8 text-base"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
          </dd>
        </div>
        <div
          class="relative"
        >
          <dt
            class="flex space-x-2"
          >
            <svg
              class="text-green-500 h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fill-rule="evenodd"
              />
            </svg>
            <h3
              class="text-xl font-medium m-0"
            >
              Application
            </h3>
          </dt>
          <dd
            class="mt-2 ml-8 text-base"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
          </dd>
        </div>
        <div
          class="relative"
        >
          <dt
            class="flex space-x-2"
          >
            <svg
              class="text-green-500 h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fill-rule="evenodd"
              />
            </svg>
            <h3
              class="text-xl font-medium m-0"
            >
              Design System
            </h3>
          </dt>
          <dd
            class="mt-2 ml-8 text-base"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
          </dd>
        </div>
      </dl>
    </div>
  </div>
</body>
`;
